### PR TITLE
Add external traffic policy to HiveMQ Platform services

### PR DESCRIPTION
**Pull request by afalhambra-hivemq**
https://hivemq.kanbanize.com/ctrl_board/22/cards/28004/details/

### CODE REVIEW

File: charts/hivemq-platform/tests/services/hivemq_control_center_services_test.yaml [hunk 1]
```diff
@@ -132,7 +132,7 @@ tests:
           hivemqProxyProtocol: true
     asserts:
       - failedTemplate:
-          errorMessage: Service type control-center with container port 8081 is using PROXY protocol value. PROXY protocol is only supported for MQTT and WebSocket services
+          errorPattern: Service type `control-center` with container port `8081` is using `hivemqProxyProtocol` value. HiveMQ PROXY protocol is only supported by MQTT and WebSocket services
```
Comment (mhofsche) [inline charts/hivemq-platform/tests/services/hivemq_control_center_services_test.yaml:135]: Why is it necessary to give the container port in this message. Should not be needed.

File: charts/hivemq-platform/tests/services/hivemq_control_center_services_test.yaml [hunk 2]
```diff
@@ -145,3 +145,26 @@ tests:
...
+  - it: with a Control Center service enabled and HiveMQ listener name enabled, validation fails
+    set:
+      services:
+        - type: control-center
+          exposed: true
+          containerPort: 8081
+          hivemqListenerName: foobar
+    asserts:
+      - failedTemplate:
+          errorPattern: Service type `control-center` with container port `8081` is using `hivemqListenerName` value. HiveMQ listener names are only supported by MQTT and WebSocket services
```
Comment (mhofsche) [inline charts/hivemq-platform/tests/services/hivemq_control_center_services_test.yaml:158]: Same here, why is the container port necessary here.

File: charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml [hunk 3]
```diff
@@ -1043,7 +1043,7 @@ tests:
           hivemqProxyProtocol: true
     asserts:
       - failedTemplate:
-          errorMessage: Service type metrics with container port 9399 is using PROXY protocol value. PROXY protocol is only supported for MQTT and WebSocket services
+          errorPattern: Service type `metrics` with container port `9399` is using `hivemqProxyProtocol` value. HiveMQ PROXY protocol is only supported by MQTT and WebSocket services
```
Comment (mhofsche) [inline charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml:1046]: Question about the use of container port in this message.

File: charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml [hunk 4]
```diff
@@ -1056,3 +1056,26 @@ tests:
...
+  - it: with a Metrics service enabled and HiveMQ listener name enabled, validation fails
+    set:
+      services:
+        - type: metrics
+          exposed: true
+          containerPort: 9399
+          hivemqListenerName: foobar
+    asserts:
+      - failedTemplate:
+          errorPattern: Service type `metrics` with container port `9399` is using `hivemqListenerName` value. HiveMQ listener names are only supported by MQTT and WebSocket services
```
Comment (mhofsche) [inline charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml:1069]: Container port?

File: charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml [hunk 5]
```diff
@@ -134,7 +134,7 @@ tests:
           hivemqProxyProtocol: true
     asserts:
       - failedTemplate:
-          errorMessage: Service type rest-api with container port 8888 is using PROXY protocol value. PROXY protocol is only supported for MQTT and WebSocket services
+          errorPattern: Service type `rest-api` with container port `8888` is using `hivemqProxyProtocol` value. HiveMQ PROXY protocol is only supported by MQTT and WebSocket services
```
Comment (mhofsche) [inline charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml:137]: Container port?

File: charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml [hunk 6]
```diff
@@ -147,3 +147,26 @@ tests:
...
+  - it: with a REST API service enabled and HiveMQ listener name enabled, validation fails
+    set:
+      services:
+        - type: rest-api
+          exposed: true
+          containerPort: 8888
+          hivemqListenerName: foobar
+    asserts:
+      - failedTemplate:
+          errorPattern: Service type `rest-api` with container port `8888` is using `hivemqListenerName` value. HiveMQ listener names are only supported by MQTT and WebSocket services
```
Comment (mhofsche) [inline charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml:160]: Container port?

File: charts/hivemq-platform/values.schema.json [hunk 7]
```diff
@@ -641,6 +620,18 @@
           "exposed" : {
             "type" : "boolean"
           },
+          "externalTrafficPolicy" : {
+            "description" : "Describes how nodes distribute service traffic they receive on one of the Service's \"externally-facing\" addresses (NodePorts and LoadBalancer).",
```
Comment (mhofsche) [inline charts/hivemq-platform/values.schema.json:624]:
```suggestion
            "description" : "Describes if the Service routes external traffic to node-local or cluster-wide endpoints. See https://kubernetes.io/docs/reference/networking/virtual-ips/#external-traffic-policy.",
```

File: charts/hivemq-platform/values.yaml [hunk 8]
```diff
@@ -257,6 +257,11 @@ services:
     #  # Session affinity type. Must be `ClientIP` or `None`.
     #  # Defaults to `ClientIP` for the `control-center` service type.
     #  type: None
+    # Controls how the network traffic is routed when the Service is accessed from an external source.
```
Comment (mhofsche) [inline charts/hivemq-platform/values.yaml:260]:
```suggestion
    # Configures whether this Service routes external traffic to node-local or cluster-wide endpoints. 
```

File: charts/hivemq-platform/tests/services/hivemq_control_center_services_test.yaml (hunk 2 above)
Comment (afalhambra-hivemq) [reply charts/hivemq-platform/tests/services/hivemq_control_center_services_test.yaml:158]: Same as explained already.

File: charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml (hunk 3 above)
Comment (afalhambra-hivemq) [reply charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml:962]: Same comment is applicable here: https://github.com/hivemq/helm-charts/pull/403#discussion_r1866225099

File: charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml (hunk 4 above)
Comment (afalhambra-hivemq) [reply charts/hivemq-platform/tests/services/hivemq_metrics_services_test.yaml:985]: Same comment is applicable here: https://github.com/hivemq/helm-charts/pull/403#discussion_r1866225099

File: charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml (hunk 6 above)
Comment (afalhambra-hivemq) [reply charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml:160]: Same comment is applicable here: https://github.com/hivemq/helm-charts/pull/403#discussion_r1866225099

File: charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml (hunk 5 above)
Comment (afalhambra-hivemq) [reply charts/hivemq-platform/tests/services/hivemq_rest_api_services_test.yaml:137]: Same comment is applicable here: https://github.com/hivemq/helm-charts/pull/403#discussion_r1866225099

**Review by Donnerbart — APPROVED** [review]

**Pull request merged**